torch/testing/_internal/opinfo/definitions/linalg.py

```diff
@@ -555,6 +558,28 @@ def sample_inputs_linalg_norm(
             ords = vector_ords if is_vector_norm else matrix_ords
 
             for ord in ords:
+                if is_vector_norm and test_size[-1] == 0:
+                    if ord == np.inf or (ord is not None and ord < 0):
+                        # RuntimeError: linalg.vector_norm cannot compute the
+                        # {ord} norm on an empty tensor because the operation
+                        # does not have an identity
+                        continue
+                elif is_matrix_norm:
+                    dims_to_check = {
+                        None: (0,),
+                        np.inf: (0,),
+                        2: (0, 1),
+                        1: (1,),
+                        -1: (1,),
+                        -2: (0, 1),
+                        -np.inf: (0,),
+                    }.get(ord, ())
```

Comment:
Are `nuc` and `fro` not considered here? And if so, why?

Reply:
The `fro` and `nuc` cases work for all sizes, so we go to the default `()` i.e. not checking any dims. I could spell it out explicitly if you prefer.

Reply:
Ah, right. For some reason I thought we were skipping that case.